galley/pkg/config/processing/snapshotter/analyzingdistributor.go

```diff
@@ -95,28 +106,37 @@ func (d *AnalyzingDistributor) Distribute(name string, s *Snapshot) {
 	// start a new analysis session
 	cancelAnalysis := make(chan struct{})
 	d.cancelAnalysis = cancelAnalysis
-	go d.analyzeAndDistribute(cancelAnalysis, s)
+	go d.analyzeAndDistribute(cancelAnalysis, name, s)
+}
+
+func (d *AnalyzingDistributor) isAnalysisSnapshot(s string) bool {
+	for _, sn := range d.s.AnalysisSnapshots {
+		if sn == s {
+			return true
+		}
+	}
 
+	return false
 }
 
-func (d *AnalyzingDistributor) analyzeAndDistribute(cancelCh chan struct{}, s *Snapshot) {
+func (d *AnalyzingDistributor) analyzeAndDistribute(cancelCh chan struct{}, name string, s *Snapshot) {
 	// For analysis, we use a combined snapshot
 	ctx := &context{
 		sn:                 d.getCombinedSnapshot(),
 		cancelCh:           cancelCh,
-		collectionReporter: d.collectionReporter,
+		collectionReporter: d.s.CollectionReporter,
 	}
 
 	scope.Analysis.Debugf("Beginning analyzing the current snapshot")
-	d.analyzer.Analyze(ctx)
+	d.s.Analyzer.Analyze(ctx)
 	scope.Analysis.Debugf("Finished analzing the current snapshot, found messages: %v", ctx.messages)
 
 	if !ctx.Canceled() {
-		d.updater.Update(ctx.messages)
+		d.s.StatusUpdater.Update(ctx.messages)
 	}
 
 	// Execution only reaches this point for default snapshot group
```

Comment:
s/default/trigger/

In any cases I regret this comment and think it's more confusing than helpful, since it sort of implies that only default gets distributed. I'm OK just yanking it entirely.